app/src/main/java/com/lucas/medicaltools/service/MainRepository.kt

```diff
@@ -0,0 +1,47 @@
+package com.lucas.medicaltools.service
+
+import android.view.View
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medical_equip.service.RetrofitBuilder
+import com.lucas.medicaltools.MainViewState
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.schedulers.Schedulers
+import retrofit2.Response
+import java.util.*
+object MainRepository {
+    fun loadUserData(): Observable<String> = Observable.just(getRandomMessage())
+
+    private fun getRandomMessage(): String {
+        val users = listOf("Tom", "Guy", "Harry")
+        return users[Random().nextInt(users.size)]
+    }
+    private fun mapMedicalTools(response: Response<List<MedicalTool>>): Observable<MainViewState> {
+        val medicalTools = response.body()
+        return if (response.isSuccessful) {
+            Observable.just(MainViewState.medicalToolsState(medicalTools))
+        } else {
+            var error = Exception(response.errorBody()?.string())
+
+            Observable.just(MainViewState.errorState(error))
+        }
+    }
+
+    fun getMedicalTools(): Observable<MainViewState>? //@NonNull io.reactivex.rxjava3.core.Observable<Observable<MainViewState>>?
```

Comment:
String for filtering comes in, you know what we discussed around what you should do depending on the string. ;)